packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -207,6 +210,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         accessibility?: N.Accessibility,
       },
       allowedModifiers: TsModifier[],
+      callback?: (startPos: number, modifer: TsModifier) => void,
```

Comment:
Alternative interface design
```ts
tsParseModifiers(
      modified: {
        [key: TsModifier]: ?true,
        accessibility?: N.Accessibility,
      },
      allowedModifiers: TsModifier[],
      disallowedModifiers: TsModifier[],
      errorTemplate: string
): void
```
So we can raise errors when a disallowed modifier is seen and it can be reused in other situations. Introducing `callback` to parsers can be very nasty.